File: zip.js
// require modules
var fs = require('fs');
var archiver = require('archiver');

// create a file to stream archive data to.
var output = fs.createWriteStream(__dirname + '/weddinggram.zip');
var archive = archiver('zip', {
    store: true // Sets the compression method to STORE.
});

// listen for all archive data to be written
output.on('close', function() {
  console.log(archive.pointer() + ' total bytes');
  console.log('archiver has been finalized and the output file descriptor has closed.');
});

// good practice to catch this error explicitly
archive.on('error', function(err) {
  throw err;
});

// pipe archive data to the file
archive.pipe(output);

// append files from a directory
archive.directory('/opt/atlassian/pipelines/agent/build/');

// finalize the archive (ie we are done appending files but streams have to finish yet)
archive.finalize();

File: zip.py
import os
import zipfile

def zipdir(path, ziph):
    # ziph is zipfile handle
    for root, dirs, files in os.walk(path):
        for file in files:
            ziph.write(os.path.join(root, file))

if __name__ == '__main__':
    zipf = zipfile.ZipFile('/opt/atlassian/pipelines/agent/build/weddinggram.zip', 'w', zipfile.ZIP_DEFLATED)
    zipdir('/opt/atlassian/pipelines/agent/build', zipf)
    zipf.close()
